server/src/main/java/org/elasticsearch/common/RunningTimeRecorder.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common;
+
+import org.elasticsearch.common.util.concurrent.ConcurrentCollections;
+import org.elasticsearch.core.Releasable;
+
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.function.LongSupplier;
+
+/**
+ * This class allows recording the total time spent running operations, it includes
+ * the elapsed time since the last reading for all ongoing recorded operations.
+ * This last property allows getting more fine-grained results and avoids skewing
+ * results for long-running operations when the sampling of this information is
+ * more frequent than the running operations.
+ */
+public class RunningTimeRecorder {
+    public static final RunningTimeRecorder NO_OP = new RunningTimeRecorder(() -> 0L) {
+        private static final Releasable EMPTY_RELEASABLE = () -> {};
+
+        @Override
+        public Releasable trackRunningTime() {
+            return EMPTY_RELEASABLE;
+        }
+
+        @Override
+        public long totalRunningTimeInNanos() {
+            return 0L;
+        }
+    };
+
+    private final LongSupplier relativeTimeSupplier;
+    private final LongAdder totalRunningTimeInNanos = new LongAdder();
+
+    private final Set<OperationRunningTimeRecorder> operationRunningTimeRecorders = ConcurrentCollections.newConcurrentSet();
```

Comment:
We only need this to be able to produce a snapshot including all ongoing operations. But since we now only need a snapshot per rollover (and not every second like in previous iterations), I think we could avoid tracking this too? Happy to keep it if you'd like to have the extra precision.